### DISCUSSION TITLE
Vercel deployment throws an error only with Safari.

### DISCUSSION THREAD

**Discussion by FrankSiyli** [neuefische/questions#575] [Web]
Hey,
my Vercel deployment throws an error: Application error: a client-side exception has occurred (see the browser console for more information).

Error shows up only with Safari on PC.

Chrome and Safari mobile is fine.

[Link to the Vercel Deployment](https://capstone-triathlon-app.vercel.app/)
[Link to the Github repo](https://github.com/FrankSiyli/capstone_triathlon_app) (main)

I tried to handle the error massages in the Safari console, but I can't find anything undefined after console.log the arrays and objects.
Also the debugger in VS Code is fine. (with my bad knowledge about debugging)
I also tried a deployment without any emojis with the same results.
I deleted the browser cache and also tried 3 different devices. Same same.

Thanks for any advice.

**Comment by toshydev** [comment]
Maybe it has to do with the deprecated method `getStorage` from `zustand/middleware`. Try to clear the local storage and then replace lines 54-56 in `store.js` with the following:
```
{
      name: "days-storage",
      storage: createJSONStorage(() => localStorage),
    }
```
Don't forget to add `createJSONStorage()` to your imports:
`import { persist, createJSONStorage } from "zustand/middleware";`

If the problem still persists, can you show the log from **vercel**?

**Edit:** Check your Safari Version. It should be Version 4+.
